tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/QueryTokenProcessor.java

```diff
@@ -0,0 +1,48 @@
+package com.google.cloud.bigquery.utils.queryfixer;
+
+import org.apache.calcite.sql.parser.babel.SqlBabelParserImpl;
+import org.apache.calcite.sql.parser.babel.Token;
+
+import com.google.cloud.bigquery.utils.queryfixer.entity.IToken;
+import com.google.cloud.bigquery.utils.queryfixer.entity.TokenImpl;
+
+import lombok.AllArgsConstructor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A processor provides methods for query and token. It can be used to tokenize queries, find tokens based on
+ * positions, and modify a query in token-level.
+ * */
+@AllArgsConstructor
+public class QueryTokenProcessor {
+
+  private final BigQueryParserFactory parserFactory;
+
+  /**
+   * tokenize a query and return all its tokens.
+   * @param sql the query to be tokenized
+   * @return a list of tokens of the query
+   */
+  public List<IToken> getAllTokens(String sql) {
+    Objects.requireNonNull(sql, "input query should not be null");
+
+    List<IToken> tokens = new ArrayList<>();
+
+    // SqlBabelParserImpl has a token manager to tokenize the input query.
+    SqlBabelParserImpl parserImpl = parserFactory.getBabelParserImpl(sql);
+
+    Token token;
+    final int EndKind = 0;
```

Comment:
If it is going to used in other places, you can make it static const of the class.